src/GraphQL.MicrosoftDI/DynamicScopedSourceStreamResolver.cs

```diff
@@ -20,11 +20,47 @@ public DynamicScopedSourceStreamResolver(ISourceStreamResolver resolver)
     {
         _resolverFunc = async context =>
         {
-            using var scope = context.RequestServicesOrThrow().CreateScope();
-            return await resolver.ResolveAsync(new ScopedResolveFieldContextAdapter<object>(context, scope.ServiceProvider)).ConfigureAwait(false);
+            var scope = context.RequestServicesOrThrow().CreateScope();
```

Comment:
This implementation is less stable in case of exceptions before newing `ObservableMapper`; `scope` variable will stay undisposed. I would place try/catch here before `new ObservableMapper(observable, scope);`

Reply:
Good catch

Reply:
Fixed